From 1f06d504365e4babb52df3ada3ac60d8a71228ee Mon Sep 17 00:00:00 2001
From: Jeremie GARCIA <jeremie.garcia@intel.com>
Date: Wed, 2 Oct 2013 15:32:56 +0200
Subject: [PDK KK AOSP]wifi: disable compilation of most of AOSP native Bluewoods ingredients

BZ: 142565

This patch disables compilation of most of AOSP native
Bluewoods ingredients.
Indeed we deals with proprietary ones except for wpa_supplicant_8_lib.

Change-Id: I94877fea8b880d40edc6afd93c53b6c2bceb8be4
Category: device enablement
Domain: CWS.WIFI-
Origin: internal
Upstream-Candidate: no, proprietary
Signed-off-by: Jeremie GARCIA <jeremie.garcia@intel.com>
---
 bcmdhd/Android.mk                           |    3 --
 bcmdhd/bcmdhd_net_iface/Android.mk          |   32 ----------------------
 bcmdhd/bcmdhd_net_iface/Android.mk.disabled |   32 ++++++++++++++++++++++
 bcmdhd/config/Android.mk                    |   39 ---------------------------
 bcmdhd/config/Android.mk.disabled           |   39 +++++++++++++++++++++++++++
 bcmdhd/dhdutil/Android.mk                   |   38 --------------------------
 bcmdhd/dhdutil/Android.mk.disabled          |   38 ++++++++++++++++++++++++++
 bcmdhd/firmware/bcm4329/device-bcm.mk       |   30 --------------------
 bcmdhd/firmware/bcm4330/device-bcm.mk       |   31 ---------------------
 bcmdhd/wpa_supplicant_8_lib/Android.mk      |    1 +
 10 files changed, 110 insertions(+), 173 deletions(-)
 delete mode 100644 bcmdhd/Android.mk
 delete mode 100644 bcmdhd/bcmdhd_net_iface/Android.mk
 create mode 100644 bcmdhd/bcmdhd_net_iface/Android.mk.disabled
 delete mode 100644 bcmdhd/config/Android.mk
 create mode 100644 bcmdhd/config/Android.mk.disabled
 delete mode 100644 bcmdhd/dhdutil/Android.mk
 create mode 100644 bcmdhd/dhdutil/Android.mk.disabled
 delete mode 100644 bcmdhd/firmware/bcm4329/device-bcm.mk
 delete mode 100644 bcmdhd/firmware/bcm4330/device-bcm.mk

diff --git a/bcmdhd/Android.mk b/bcmdhd/Android.mk
deleted file mode 100644
index 5b43128..0000000
--- a/bcmdhd/Android.mk
+++ /dev/null
@@ -1,3 +0,0 @@
-ifeq ($(BOARD_WLAN_DEVICE),bcmdhd)
-    include $(call all-subdir-makefiles)
-endif
diff --git a/bcmdhd/bcmdhd_net_iface/Android.mk b/bcmdhd/bcmdhd_net_iface/Android.mk
deleted file mode 100644
index c5dc34d..0000000
--- a/bcmdhd/bcmdhd_net_iface/Android.mk
+++ /dev/null
@@ -1,32 +0,0 @@
-#
-# Copyright (C) 2012 Google, inc.
-#
-# $Id: Android.mk,v 2.6 2009-05-07 18:25:15 hharte Exp $
-#
-# Permission to use, copy, modify, and/or distribute this software for any
-# purpose with or without fee is hereby granted, provided that the above
-# copyright notice and this permission notice appear in all copies.
-#
-# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
-# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
-# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
-# SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
-# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
-# OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
-# CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
-#
-
-#
-# lib_net_iface_cmd
-#
-
-LOCAL_PATH := $(call my-dir)
-include $(CLEAR_VARS)
-LOCAL_SRC_FILES := bcmdhd_net_iface.c
-LOCAL_MODULE := libnetcmdiface
-ifeq ($(TARGET_ARCH),arm)
-LOCAL_CFLAGS := -mabi=aapcs-linux
-endif
-LOCAL_MODULE_TAGS := optional
-LOCAL_ALLOW_UNDEFINED_SYMBOLS := true
-include $(BUILD_SHARED_LIBRARY)
diff --git a/bcmdhd/bcmdhd_net_iface/Android.mk.disabled b/bcmdhd/bcmdhd_net_iface/Android.mk.disabled
new file mode 100644
index 0000000..c5dc34d
--- /dev/null
+++ b/bcmdhd/bcmdhd_net_iface/Android.mk.disabled
@@ -0,0 +1,32 @@
+#
+# Copyright (C) 2012 Google, inc.
+#
+# $Id: Android.mk,v 2.6 2009-05-07 18:25:15 hharte Exp $
+#
+# Permission to use, copy, modify, and/or distribute this software for any
+# purpose with or without fee is hereby granted, provided that the above
+# copyright notice and this permission notice appear in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
+# SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
+# OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+# CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+#
+
+#
+# lib_net_iface_cmd
+#
+
+LOCAL_PATH := $(call my-dir)
+include $(CLEAR_VARS)
+LOCAL_SRC_FILES := bcmdhd_net_iface.c
+LOCAL_MODULE := libnetcmdiface
+ifeq ($(TARGET_ARCH),arm)
+LOCAL_CFLAGS := -mabi=aapcs-linux
+endif
+LOCAL_MODULE_TAGS := optional
+LOCAL_ALLOW_UNDEFINED_SYMBOLS := true
+include $(BUILD_SHARED_LIBRARY)
diff --git a/bcmdhd/config/Android.mk b/bcmdhd/config/Android.mk
deleted file mode 100644
index b125897..0000000
--- a/bcmdhd/config/Android.mk
+++ /dev/null
@@ -1,39 +0,0 @@
-#
-# Copyright (C) 2008 The Android Open Source Project
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
-LOCAL_PATH := $(call my-dir)
-
-########################
-
-include $(CLEAR_VARS)
-LOCAL_MODULE := dhcpcd.conf
-LOCAL_MODULE_CLASS := ETC
-LOCAL_MODULE_PATH := $(TARGET_OUT_ETC)/dhcpcd
-LOCAL_SRC_FILES := android_dhcpcd.conf
-include $(BUILD_PREBUILT)
-
-#########################
-
-WIFI_DRIVER_SOCKET_IFACE := wlan0
-ifeq ($(strip $(WPA_SUPPLICANT_VERSION)),VER_0_8_X)
-  include external/wpa_supplicant_8/wpa_supplicant/wpa_supplicant_conf.mk
-else
-ifeq ($(strip $(WPA_SUPPLICANT_VERSION)),VER_0_6_X)
-  include external/wpa_supplicant_6/wpa_supplicant/wpa_supplicant_conf.mk
-else
-  include external/wpa_supplicant/wpa_supplicant_conf.mk
-endif
-endif
-#######################
diff --git a/bcmdhd/config/Android.mk.disabled b/bcmdhd/config/Android.mk.disabled
new file mode 100644
index 0000000..b125897
--- /dev/null
+++ b/bcmdhd/config/Android.mk.disabled
@@ -0,0 +1,39 @@
+#
+# Copyright (C) 2008 The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+LOCAL_PATH := $(call my-dir)
+
+########################
+
+include $(CLEAR_VARS)
+LOCAL_MODULE := dhcpcd.conf
+LOCAL_MODULE_CLASS := ETC
+LOCAL_MODULE_PATH := $(TARGET_OUT_ETC)/dhcpcd
+LOCAL_SRC_FILES := android_dhcpcd.conf
+include $(BUILD_PREBUILT)
+
+#########################
+
+WIFI_DRIVER_SOCKET_IFACE := wlan0
+ifeq ($(strip $(WPA_SUPPLICANT_VERSION)),VER_0_8_X)
+  include external/wpa_supplicant_8/wpa_supplicant/wpa_supplicant_conf.mk
+else
+ifeq ($(strip $(WPA_SUPPLICANT_VERSION)),VER_0_6_X)
+  include external/wpa_supplicant_6/wpa_supplicant/wpa_supplicant_conf.mk
+else
+  include external/wpa_supplicant/wpa_supplicant_conf.mk
+endif
+endif
+#######################
diff --git a/bcmdhd/dhdutil/Android.mk b/bcmdhd/dhdutil/Android.mk
deleted file mode 100644
index 4294413..0000000
--- a/bcmdhd/dhdutil/Android.mk
+++ /dev/null
@@ -1,38 +0,0 @@
-#
-# Copyright (C) 2008-2011 Broadcom Corporation
-#
-# $Id: Android.mk,v 2.6 2009-05-07 18:25:15 $
-#
-# Permission to use, copy, modify, and/or distribute this software for any
-# purpose with or without fee is hereby granted, provided that the above
-# copyright notice and this permission notice appear in all copies.
-#
-# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
-# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
-# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
-# SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
-# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
-# OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
-# CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
-#
-
-LOCAL_PATH:= $(call my-dir)
-
-include $(CLEAR_VARS)
-LOCAL_SRC_FILES := \
-	dhdu.c \
-	dhdu_linux.c \
-	bcmutils.c \
-	miniopt.c
-
-LOCAL_MODULE := dhdutil
-LOCAL_CFLAGS := -DSDTEST -DTARGETENV_android -Dlinux -DLINUX
-ifeq ($(TARGET_ARCH),arm)
-LOCAL_CFLAGS += -mabi=aapcs-linux
-endif
-LOCAL_C_INCLUDES +=$(LOCAL_PATH)/include
-
-LOCAL_MODULE_PATH := $(TARGET_OUT_OPTIONAL_EXECUTABLES)
-LOCAL_MODULE_TAGS := debug
-
-include $(BUILD_EXECUTABLE)
diff --git a/bcmdhd/dhdutil/Android.mk.disabled b/bcmdhd/dhdutil/Android.mk.disabled
new file mode 100644
index 0000000..4294413
--- /dev/null
+++ b/bcmdhd/dhdutil/Android.mk.disabled
@@ -0,0 +1,38 @@
+#
+# Copyright (C) 2008-2011 Broadcom Corporation
+#
+# $Id: Android.mk,v 2.6 2009-05-07 18:25:15 $
+#
+# Permission to use, copy, modify, and/or distribute this software for any
+# purpose with or without fee is hereby granted, provided that the above
+# copyright notice and this permission notice appear in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
+# SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
+# OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+# CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+#
+
+LOCAL_PATH:= $(call my-dir)
+
+include $(CLEAR_VARS)
+LOCAL_SRC_FILES := \
+	dhdu.c \
+	dhdu_linux.c \
+	bcmutils.c \
+	miniopt.c
+
+LOCAL_MODULE := dhdutil
+LOCAL_CFLAGS := -DSDTEST -DTARGETENV_android -Dlinux -DLINUX
+ifeq ($(TARGET_ARCH),arm)
+LOCAL_CFLAGS += -mabi=aapcs-linux
+endif
+LOCAL_C_INCLUDES +=$(LOCAL_PATH)/include
+
+LOCAL_MODULE_PATH := $(TARGET_OUT_OPTIONAL_EXECUTABLES)
+LOCAL_MODULE_TAGS := debug
+
+include $(BUILD_EXECUTABLE)
diff --git a/bcmdhd/firmware/bcm4329/device-bcm.mk b/bcmdhd/firmware/bcm4329/device-bcm.mk
deleted file mode 100644
index ef42698..0000000
--- a/bcmdhd/firmware/bcm4329/device-bcm.mk
+++ /dev/null
@@ -1,30 +0,0 @@
-#
-# Copyright (C) 2008 The Android Open Source Project
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
-
-########################
--include hardware/broadcom/wlan/bcmdhd/config/config-bcm.mk
-
-ifeq ($(strip $(WIFI_BAND)),802_11_ABG)
-BCM_FW_SRC_FILE_STA := fw_bcm4329_abg.bin
-else
-BCM_FW_SRC_FILE_STA := fw_bcm4329.bin
-endif
-BCM_FW_SRC_FILE_AP  := fw_bcm4329_apsta.bin
-
-PRODUCT_COPY_FILES += \
-    hardware/broadcom/wlan/bcmdhd/firmware/bcm4329/$(BCM_FW_SRC_FILE_STA):system/vendor/firmware/fw_bcmdhd.bin \
-    hardware/broadcom/wlan/bcmdhd/firmware/bcm4329/$(BCM_FW_SRC_FILE_AP):system/vendor/firmware/fw_bcmdhd_apsta.bin
-########################
diff --git a/bcmdhd/firmware/bcm4330/device-bcm.mk b/bcmdhd/firmware/bcm4330/device-bcm.mk
deleted file mode 100644
index d9acd67..0000000
--- a/bcmdhd/firmware/bcm4330/device-bcm.mk
+++ /dev/null
@@ -1,31 +0,0 @@
-#
-# Copyright (C) 2008 The Android Open Source Project
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
-
-########################
--include hardware/broadcom/wlan/bcmdhd/config/config-bcm.mk
-
-ifeq ($(strip $(WIFI_BAND)),802_11_BG)
-BCM_FW_SRC_FILE_STA := fw_bcm4330_bg.bin
-BCM_FW_SRC_FILE_AP  := fw_bcm4330_apsta_bg.bin
-else
-BCM_FW_SRC_FILE_STA := fw_bcm4330_abg.bin
-BCM_FW_SRC_FILE_AP  := fw_bcm4330_apsta_bg.bin
-endif
-
-PRODUCT_COPY_FILES += \
-    hardware/broadcom/wlan/bcmdhd/firmware/bcm4330/$(BCM_FW_SRC_FILE_STA):system/vendor/firmware/fw_bcmdhd.bin \
-    hardware/broadcom/wlan/bcmdhd/firmware/bcm4330/$(BCM_FW_SRC_FILE_AP):system/vendor/firmware/fw_bcmdhd_apsta.bin
-########################
diff --git a/bcmdhd/wpa_supplicant_8_lib/Android.mk b/bcmdhd/wpa_supplicant_8_lib/Android.mk
index 146e04d..8381dd6 100644
--- a/bcmdhd/wpa_supplicant_8_lib/Android.mk
+++ b/bcmdhd/wpa_supplicant_8_lib/Android.mk
@@ -56,6 +56,7 @@ endif
 
 include $(CLEAR_VARS)
 LOCAL_MODULE := lib_driver_cmd_bcmdhd
+LOCAL_MODULE_TAGS := optional
 LOCAL_SHARED_LIBRARIES := libc libcutils
 LOCAL_CFLAGS := $(L_CFLAGS)
 LOCAL_SRC_FILES := $(WPA_SRC_FILE)
-- 
1.7.4.1

